scipy/sparse/_index.py

```diff
@@ -266,8 +266,13 @@ def _unpack_index(index):
         else:
             raise IndexError('invalid number of indices')
     else:
-        row, col = index, slice(None)
```

Comment:
Earlier `col` was set to `None` for non `numpy.ndarray`  irrespective of the dimension, which is why the 2D index array couldn't have worked.